Lesson15/havva/script.js

```diff
@@ -0,0 +1,223 @@
+  `);
+
+
+// 4. Use template literals to log: "My name is (userName). I like JS."
+// CODE HERE
+
+console.log(`My name is ${userName}. I like JS.`);
+
+
+// 5. Declare a string password = "securePass". Log the length of password.
+// CODE HERE
+
+const securePass = '1234567';
+
+console.log('The length of the password is:', securePass.length);
+
+
+
+// 6. Convert the string "hello world" to uppercase and log it.
+// CODE HERE
+const sentence = 'Hello world';
+console.log(sentence.toUpperCase(), sentence);
+
+
+// 7. Concatenate "Hello" and "World" with a space in between and log the result.
+// CODE HERE
+console.log("Hello" + " " + "World");
+
+
+// 8. Check the type of a variable, e.g., let x = 42. Log the type using typeof.
+// CODE HERE
+const x = 42 + 10 + 10;
+const y = x + '10';
+console.log('type of x is: ', typeof x, x, typeof y, y);
+
+
+// 9. Convert the number 100 to a string and log the result.
+// CODE HERE
+const numberToString = 100;
+const convertedString = String(numberToString);
+// Alternatively, you can use numberToString.toString();
+console.log('Converted number to string:', convertedString, 'Type:', typeof convertedString);
+
+// 10. Convert the string "50" to a number and log its type to confirm the conversion.
+// CODE HERE
+
+const stringToNumber = '50';
+const convertedNumber = Number(stringToNumber);
+// Alternatively, you can use parseInt(stringToNumber) or parseFloat(stringToNumber);
+// const convertedNumber = parseInt(stringToNumber, 10); // Using parseInt with base 10
+console.log('Converted string to number:', convertedNumber, 'Type:', typeof convertedNumber);
+
+
+// 11. Generate a random integer between 2 and 10 and log it.
+// 0 8 => from 2 to 10
+const randomNumber = Math.floor(Math.random() * 9) + 2;
+
+console.log('random number is', randomNumber);
+
+
+// 12. Round the number 3.7 down using Math.floor and 3.2 up using Math.ceil, log both.
+// CODE HERE
+
+const number1 = 3.7;
+const number2 = 3.2;
+const roundedDown = Math.floor(number1);
+const roundedUp = Math.ceil(number2);
+console.log('Rounded down:', roundedDown, 'Rounded up:', roundedUp);
+
+// 13. Declare a boolean variable isStudent = true. Log it.
+// CODE HERE
+
+const isStudent = true;
+console.log('Is student:', isStudent);
+
+// 14. Initialize counter = 0, then increment it by 1 using counter++ and log it.
+// CODE HERE
+
+let counter = 0;
+counter++;
+console.log('Counter after increment:', counter);
+
+// 15. Initialize points = 10, add 5 to it using points += 5, then log points.
+// CODE HERE
+
+let points = 10;
+points += 5;
+console.log('Points after addition:', points);
+
+// 16. Declare name="Alice", age=30, city="Paris". Log "Alice (30) lives in Paris" using template literals.
+// CODE HERE
+
+const name = 'Alice';
+const age = 30;
+const city = 'Paris';
+console.log(`${name} (${age}) lives in ${city}`);
+
+// 17. Declare variables x=5, y=10, z=15. Log their total sum.
+// CODE HERE
+
+const x1 = 5;
```

Comment:
Well done here with the name. Since we write all the code in the same file (without using functions or other code blocks), everything is in the so-called global scope. This means that we cannot create more than one variable with the same name. 👍